Python/ceval.c

```diff
@@ -2124,6 +2124,19 @@ _PyEval_UnpackIterableStackRef(PyThreadState *tstate, _PyStackRef v_stackref,
         _PyErr_Format(tstate, PyExc_ValueError,
                       "too many values to unpack (expected %d)",
                       argcnt);
+        PyObject *exc = _PyErr_GetRaisedException(tstate);
+        if (PyObject_SetAttr(exc, &_Py_ID(_unpacked_value), v)) {
+            _PyErr_ChainExceptions1(exc);
```

Comment:
If `PyObject_SetAttr` fails, should we still chain the exception?

Reply:
IMO, chaining is fine.
We know that `exc` is a `ValueError`, so a failed setattr should be very rare.